src/platform/unix/mod.rs

```diff
@@ -105,11 +105,20 @@ pub unsafe fn init_os_handler() -> Result<(), Error> {
     };
 
     #[cfg(feature = "termination")]
-    match signal::sigaction(signal::Signal::SIGTERM, &new_action) {
-        Ok(_) => {}
-        Err(e) => {
-            signal::sigaction(signal::Signal::SIGINT, &_old).unwrap();
-            return Err(close_pipe(e));
+    {
+        match signal::sigaction(signal::Signal::SIGTERM, &new_action) {
+            Ok(_) => {}
+            Err(e) => {
+                signal::sigaction(signal::Signal::SIGINT, &_old).unwrap();
+                return Err(close_pipe(e));
+            }
+        }
+        match signal::sigaction(signal::Signal::SIGHUP, &new_action) {
+            Ok(_) => {}
+            Err(e) => {
+                signal::sigaction(signal::Signal::SIGINT, &_old).unwrap();
```

Comment:
Old value for SIGTERM should be reverted in this cleanup function. Also, the variable `_old` should be named `old` now as it is being used. Maybe assign current `_old` to `sigint_old` and introduce `sigterm_old`.

Reply:
I made a change. Is that maybe what you have in mind?